spark/src/main/scala/io/spark/ddf/util/SparkUtils.scala

```diff
@@ -119,6 +124,74 @@ object SparkUtils {
 
   }
 
+  /**
+   *
+   * @param df
+   * @return an Array of string showing the dataframe with complex column-object replaced by json string
+   */
+  def jsonForComplexType(df: DataFrame, sep: String): Array[String] = {
+    val schema = df.schema
+    val df1: RDD[String] = df.map(r => rowToJSON(schema, r, sep))
+    df1.collect()
+  }
+
+  private def rowToJSON(rowSchema: StructType, row: Row, separator: String): String = {
```

Comment:
don't need to do this, use https://github.com/apache/spark/blob/branch-1.3/sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala#L1236